src/sonic_ax_impl/lib/quaggaclient.py

```diff
@@ -99,16 +100,27 @@ def union_bgp_sessions(self):
         sessions_ipv6 = parse_bgp_summary(bgpsumm_ipv6)
 
         ## Note: sessions_ipv4 will overwrite sessions_ipv6 if key is the same
-        sessions = {}
+        neighbor_sessions = {}
         for ses in sessions_ipv6 + sessions_ipv4:
             nei = ses['Neighbor']
-            sessions[nei] = ses
-        return sessions
+            neighbor_sessions[nei] = ses
+        return neighbor_sessions
 
     def auth(self):
         cmd = b"zebra\n"
         self.sock.send(cmd)
-        return self.vtysh_recv()
+
+        ## Nowadays we see 2 BGP stack
+        ## 1. Quagga (version 0.99.24.1)
+        ## 2. FRRouting (version 7.2-sonic)
+        banner = self.vtysh_recv()
+        if 'Quagga' in banner:
+            self.bpg_provider = 'Quagga'
+        elif 'FRRouting' in banner:
+            self.bpg_provider = 'FRRouting'
+        else:
```

Comment:
I'd suggest to use json output in case of FRR. FRR supports it and also FRR output could be changed,

Reply:
Will do in future PR.

---
In reply to: [356096823](https://github.com/Azure/sonic-snmpagent/pull/119#discussion_r356096823) [](ancestors = 356096823)